test/priors.cpp

```diff
@@ -0,0 +1,51 @@
+#include <gtest/gtest.h>
+#include <memory>
+
+//#include <Eigen/Dense>
+// #include <stan/math/prim/fun.hpp>
+// #include <stan/math/prim/prob.hpp>
+
+//#include "../proto/cpp/marginal_state.pb.h"
+#include "../src/hierarchies/base_hierarchy.hpp"
+//#include "../src/hierarchies/nnig_hierarchy.hpp"
+#include "../src/mixings/dirichlet_mixing.hpp"
+
+
+TEST(mixing, fixed_value) {
+  DirichletMixing mix;
+  bayesmix::DPState state;
+  double m = 2.0;
+  state.mutable_fixed_value()->set_value(m);
+  double m_state = state.fixed_value().value();
+  ASSERT_FLOAT_EQ(m, m_state);
+  mix.set_state(&state);
+  double m_mix = mix.get_totalmass();
+  ASSERT_FLOAT_EQ(m, m_mix);
+
+  std::vector<std::shared_ptr<BaseHierarchy>> hiers(5);
+  mix.update_hypers(hiers, 100);
+  double m_mix_after = mix.get_totalmass();
+  ASSERT_FLOAT_EQ(m, m_mix_after);
+}
+
+
+
+TEST(mixing, gamma_prior) {
+  DirichletMixing mix;
+  bayesmix::DPState state;
+  double alpha = 1.0;
+  double beta = 2.0;
+  double m_prior = alpha / beta;
+  state.mutable_gamma_prior()->set_alpha(alpha);
+  state.mutable_gamma_prior()->set_beta(beta);
+  mix.set_state(&state);
+  double m_mix = mix.get_totalmass();
+  ASSERT_FLOAT_EQ(m_prior, m_mix);
+
+  std::vector<std::shared_ptr<BaseHierarchy>> hiers(5);
+  mix.update_hypers(hiers, 100);
+  double m_mix_after = mix.get_totalmass();
+
+  std::cout << "[          ] after = " << m_mix_after << std::endl;
+  ASSERT_TRUE(m_mix != m_mix_after);
```

Comment:
maybe we could be more specific: with 100 clusters and 1000 samples I would expect m_mix_after to be greater than m_mix ?

Reply:
I don't know, would I? What's the reasoning behind it?

Reply:
Well that's the meaning of totalmass: it should quantify the number of clusters (a priori)

So that if we start with totalmass small (a priori) and then observe a lot of clusters, I would expect that a posteriori we should have larger totalmass